### ISSUE TITLE
Can't compile rippled with BOOST_ROOT set

### ISSUE THREAD

**Issue by donovanhide** [ripple/rippled#3384]
Having followed the installation instructions here:

https://xrpl.org/build-run-rippled-ubuntu.html

On Ubuntu Focal with boost 1.71 compiled, as per instructions, I get the following output:

```
$ echo $BOOST_ROOT
/home/admin/boost_1_71_0
$ pwd
/home/admin/rippled/build
$ cmake ..
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.25.1")
-- Using 2 cores for ExternalProject builds.
-- Build type not specified - defaulting to Release
-- rippled version: 1.5.0
-- NIH-EP cache path: /home/admin/rippled/.nih_c/unix_makefiles/GNU_9.3.0/Release
-- docker NOT found -- won't be able to build containers for packaging
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- BOOST_ROOT: /home/admin/boost_1_71_0
-- BOOST_LIBRARYDIR:
CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake:117 (find_package):
  Found package configuration file:

    /usr/lib/x86_64-linux-gnu/cmake/boost_chrono-1.71.0/boost_chrono-config.cmake

  but it set boost_chrono_FOUND to FALSE so package "boost_chrono" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.

  The following variants have been tried and rejected:

  * libboost_chrono.so.1.71.0 (shared, Boost_USE_STATIC_LIBS=ON)

  * libboost_chrono.a (shared runtime, Boost_USE_STATIC_RUNTIME=ON)

Call Stack (most recent call first):
  /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake:182 (boost_find_component)
  Builds/CMake/deps/FindBoost.cmake:273 (find_package)
  Builds/CMake/deps/Boost.cmake:41 (find_package)
  CMakeLists.txt:44 (include)


-- Configuring incomplete, errors occurred!
See also "/home/admin/rippled/build/CMakeFiles/CMakeOutput.log".
See also "/home/admin/rippled/build/CMakeFiles/CMakeError.log".

```

**Comment by donovanhide** [comment]
Looks like it has something to do with:

/usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake

**Comment by carlhua** [comment]
It seems to me that your build of boost cant find the proper packages. I suggest you re-build boost from scratch and give it a try. This is not a rippled issue - closing issue. Thank you

**Comment by donovanhide** [comment]
That was a bit abrupt and unhelpful.

**Comment by carlhua** [comment]
@donovanhide I closed the issue because rippled is passing CI and I have just built it this morning myself. So its not an issue of rippled. However I am happy to see if I can help - can you try following the instructions here: https://github.com/ripple/rippled/blob/develop/Builds/linux/README.md and let me know if it works?

**Comment by donovanhide** [comment]
I linked to the more up to date instructions that I followed in the original report above. The problem is to do with the ubuntu version of cmake combined with the ubuntu version of the boost libraries, which is not directly linked to rippled, but it is likely that users of rippled will encounter if they choose to compile on a machine that they have compiled previous versions of rippled and boost on. This obviously isn't the case on a CI box.
